### PR TITLE
Move GitHub workflow dependency installation to its own action file

### PR DESCRIPTION
**Pull request by dannycjones**
This change collects all the dependency installation into one script, such that it is consistent across operating systems.

This should allow us to start building a matrix up for different operating systems, architectures, AWS regions, etc..

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

**Comment by dannycjones** [comment]
Note, CI isn't going to actually test these changes since it won't pull in the updated workflow spec.

**Comment by sauraank** [comment]
> Note, CI isn't going to actually test these changes since it won't pull in the updated workflow spec.
> 
> Here's a push to my fork which shows the Ubuntu runner working - I can't test self-hosted runner right now. https://github.com/dannycjones/mountpoint-s3/actions/runs/5315330797/jobs/9623572410

How do you test the runner?

File: doc/BENCHMARKING.md [hunk 1]
```diff
@@ -22,12 +22,11 @@ We keep the records of benchmarking results in `gh-pages` branch and the perform
 While our benchmark script is written for CI testing only, it is possible to run manually.
 You can use the following steps.
 
-1. Install dependencies by running below commands for Ubuntu.
+1. Install dependencies and configure FUSE by running the following script in the repository:
```
Comment (sauraank) [inline doc/BENCHMARKING.md:25]: apt-get does not work outside Linux, right?

**Comment by dannycjones** [comment]
> How do you test the runner?

For the GitHub managed runner, I've just pushed to my fork's main branch and seen all is OK.

For self-hosted runners, we'd have to push and find out later 😄 That being said, I did launch an EC2 Instance and run the commands by hand and check tests pass as expected.

**Review by sauraank — COMMENTED** [review]
How were Github Workflow dependencies installed earler?

File: doc/BENCHMARKING.md (hunk 1 above)
Comment (dannycjones) [reply doc/BENCHMARKING.md:25]: It won't work outside of Debian-based operating systems (so Ubuntu, Debian).

**Comment by dannycjones** [comment]
>How were Github Workflow dependencies installed earler?

They were installed just using `apt-get` or `yum` hardcoded for each job. This takes those steps and puts them in a script, which figures out which package manager to use and which package names to use if they are different.

File: doc/BENCHMARKING.md (hunk 1 above)
Comment (dannycjones) [reply doc/BENCHMARKING.md:25]: Also, the script only works on distros using apt/apt-get and rpm/yum/dnf.

**Review by sauraank — APPROVED** [review]

**Pull request merged**